### PR TITLE
Signup Button Click

### PR DESCRIPTION
**Pull request by pamelaabreu**

### CODE REVIEW

File: src/containers/Signup/Signup.js [hunk 1]
```diff
@@ -1,40 +1,55 @@
...
 
-    render () {
+    validateFormInputs = () => {
         const { inputs } = this.state;
-        const inputArray = Object.entries(inputs);
+        
+        return inputs.email.length > 0 && inputs.password.length > 0;
+    }
+
+    handleSignupButtonClick = e => {
```
Comment (JoshuaKGoldberg) [inline src/containers/Signup/Signup.js]: You don't have to do this in JavaScript: 

* General guide: https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation
* Example docs, for number inputs: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#Validation

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
You'll have an easier time with these kinds of validations if you use HTML attributes when possible. See the comment in `Signup.js`.

Also, a question: is this really necessary for your MVP? Do you have to validate user inputs, etc. to showcase in a demo that this works?

Since you're on a tight time budget, I'd suggest not worrying about these kinds of _"what if the user does the wrong thing?"_ concerns. Just focus on what benefits you to show in the demo, and file a note to do these later.

**Comment by JoshuaKGoldberg** [comment]
Btw, if you disagree with one of my reviews, you can always dismiss it or re-request review in the UI. I'm using it to show how I would review this at work, but you definitely shouldn't be blocked on me being wrong about something. 😄 

image

File: src/containers/Signup/Signup.js [hunk 2]
```diff
@@ -1,42 +1,73 @@
...
+      });
+  };
+
+  render() {
+    const { inputs, fillFormInputError, firebaseCreateUserError } = this.state;
+    const inputsArray = Object.entries(inputs);
+
+    return (
+      <form onSubmit={this.handleSignupSubmit}>
+        {fillFormInputError ? <p>{fillFormInputError}</p> : null}
```
Comment (JoshuaKGoldberg) [inline src/containers/Signup/Signup.js]: `fillFormInputError` is never assigned to, so you can get rid of it from here and in `state`.

File: src/containers/Signup/Signup.js [hunk 3]
```diff
@@ -1,42 +1,73 @@
...
+        {firebaseCreateUserError ? <p>{firebaseCreateUserError}</p> : null}
+        {inputsArray.map(([inputName, inputValue], index) => {
+          const inputType =
+            inputName.toLowerCase() === "password" ? "password" : "text";
+          const isRequired =
+            inputName.toLowerCase() === "username" ? false : true;
```
Comment (JoshuaKGoldberg) [inline src/containers/Signup/Signup.js]: Nitpicking here: you can replace `=== "username" ? false : true` with `!== "username"`

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM! Just a couple of cleanup comments.

**Comment by FiveEightyEight** [comment]
Thank you for the reviews Josh!

**Pull request merged**